ReleaseProcessScript/tools/PesterTester/semver_functions.Tests.ps1

```diff
@@ -95,6 +95,34 @@ Describe "semver_functions" {
 
       Get-Possible-Versions-Hotfix $Version | Should Be $NextVersions
     }
+
+    It "Get-Possible-Next-Versions-Hotfix_WithAlpha_NotMeantForCurrentVersion_ShouldReturnArray" {
+      $Version = "2.28.0-alpha.1"
+      $NextVersions = "2.28.0-alpha.2", "2.28.0-beta.1", "2.28.1"
```

Comment:
Shouldn't `2.28.0-rc.1` also be part of this? `Get-Possible-Next-Versions-Hotfix_WithRc_NotMeantForCurrentVersion_ShouldReturnArray` shows the possibility of `rc`'s on hotfix branches (I guess?). @MichaelKetting ?

Reply:
that would be a separate issue. 

As to the reasoning about rc pro/contra: we should never want to start an rc with open issues to be fixed in the rc. For practical purposes, it might be helpful, on occasion. I still hope it's uncommon enough a scenario that it won't matter so there's no need to add an issue for this.

Reply:
I have refactored the method such that it (hopefully) has a better naming and the initially extracted version part variables (from the regex) are not mutated.